flytekit/annotated/dynamic_mixin.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, Callable, Union
+
+from flytekit.annotated.context_manager import ExecutionState, FlyteContext
+from flytekit.annotated.workflow import Workflow, WorkflowFailurePolicy, WorkflowMetadata, WorkflowMetadataDefaults
+from flytekit.loggers import logger
+from flytekit.models import dynamic_job as _dynamic_job
+from flytekit.models import literals as _literal_models
+
+
+class DynamicWorkflowTaskMixin(object):
```

Comment:
why keep this separate? Whats the use of a mixin? We will never use it elsewhere right?

Reply:
if you want you can just make these has helper methods?

Reply:
nope, see my comment above I wasn't sure whether it was cleaner to leave PythonFunctionTask as is given it's so simple. I'll fold this in.